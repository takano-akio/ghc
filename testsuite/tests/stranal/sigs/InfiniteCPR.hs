module InfiniteCPR where

data Rec1 = Foo Rec2 Rec2
data Rec2 = Bar Rec1 Rec1

f a =
    let x = Foo a y
        y = Bar x x
    in x

