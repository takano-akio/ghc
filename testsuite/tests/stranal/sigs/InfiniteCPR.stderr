
==================== Strictness signatures ====================
InfiniteCPR.$tc'Bar: tm(t,t,tm(tm1,tm1),tm1,t,tm4)
InfiniteCPR.$tc'Foo: tm(t,t,tm(tm1,tm1),tm1,t,tm4)
InfiniteCPR.$tcRec1: tm(t,t,tm(tm1,tm1),tm1,t,tm5)
InfiniteCPR.$tcRec2: tm(t,t,tm(tm1,tm1),tm1,t,tm5)
InfiniteCPR.$trModule: tm(tm1,tm1)
InfiniteCPR.f: <L,U>m(,tm(m(,tm(,)),m(,tm(,))))



==================== Strictness signatures ====================
InfiniteCPR.$tc'Bar: tm(t,t,tm(tm1,tm1),tm1,t,tm4)
InfiniteCPR.$tc'Foo: tm(t,t,tm(tm1,tm1),tm1,t,tm4)
InfiniteCPR.$tcRec1: tm(t,t,tm(tm1,tm1),tm1,t,tm5)
InfiniteCPR.$tcRec2: tm(t,t,tm(tm1,tm1),tm1,t,tm5)
InfiniteCPR.$trModule: tm(tm1,tm1)
InfiniteCPR.f: <L,U>m(,tm(m(,tm(,)),m(,tm(,))))


